Eatery/Views/Eateries/EateryCollectionViewCell.swift

```diff
@@ -0,0 +1,332 @@
+import UIKit
+import CoreLocation
+import Kingfisher
+import SnapKit
+
+class EateryCollectionViewCell: UICollectionViewCell {
+
+    private static let shadowRadius: CGFloat = 12
+
+    let paymentContainer: UIView
+    let paymentImageViews: [UIImageView]
+
+    let backgroundImageView: UIImageView
+    let closedOverlay: UIView
+
+    let infoContainer: UIView
+    let titleLabel: UILabel
+    let statusLabel: UILabel
+    let timeLabel: UILabel
+    let distanceLabel: UILabel
+
+    let menuTextView: UITextView
+    private var menuTextViewHiddenConstraints: [Constraint]
+    private var menuTextViewVisibleConstraints: [Constraint]
+
+    var isMenuTextViewVisible: Bool = false {
+        didSet {
+            if isMenuTextViewVisible {
+                showMenuTextView()
+            } else {
+                hideMenuTextView()
+            }
+        }
+    }
+
+    var eatery: Eatery? {
+        didSet {
+            updateInfoViews()
+        }
+    }
+
+    var userLocation: CLLocation? {
+        didSet {
+            updateDistanceLabelText()
+        }
+    }
+
+    override init(frame: CGRect) {
```

Comment:
this is a giant init func. Should probably break it up into some smaller functions to increase readability